### ISSUE TITLE
Primary route triggers change when auxiliary route is triggered on "/"

### ISSUE THREAD

**Issue by Epenance** [angular/angular#21966]
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report 
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
</code></pre>

## Current behavior
If you have two router outlets in a given app, one "primary" router-outlet and one auxiliary router-outlet (named outlet), and standing on the root of the domain and triggers a auxiliary route with the routerLink, the browser will insert a "/" before the (auxiliary:route) in the url, this causes a change detection in the router.

If you use a no reuse strategy and wildcard routes (Needed for hooking up to a CMS), the primary route will be rerendered and therefore triggering any http calls, animations and such there may be associated with it.

However if you stand on a subpage in the primary outlet, there is no issue since the "/" will not be inserted before the (auxiliary:route) in the url and therefore not triggering a change event.

Example: https://stackblitz.com/edit/angular-3rhhpw

## Expected behavior
I would expect the primary router not to be changed if you are on the root of the domain, and have the same behaviour as any other subpage.

## Minimal reproduction of the problem with instructions
Example: https://angular-3rhhpw.stackblitz.io/list1

## What is the motivation / use case for changing the behavior?
I'm using the auxiliary routes to open an overlay and display some data to make it deeplinkable within my application. The content is coming from a CMS and therefore we use the wildcard routes and noreuse strategy for the route components.

## Environment

<pre><code>
Angular version: ^5.2.1 & 5.0.0

Browser:
- [x] Chrome (desktop) version XX
- [x] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [x] Firefox version XX
- [x] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [x] IE version XX
- [x] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by Epenance** [comment]
Okay, so I seem to have found a work around and gotten closer to finding out what causes the bug.

As explained i use a `**`wildcard route for everything and apparently when opening a auxiliary route the wildcard route does not pick up the `/`route, however adding a new empty route with the exact same configuration as the `**` fixes this and it recognizes the route even when the other is open.

Example: https://stackblitz.com/edit/angular-v9hptr

**Comment by jasonaden** [comment]
This issue is a bit hard to follow. Maybe you can produce clearer steps to reproduce, such as: 

1. Go to XYZ url
2. Click on ABC link and notice URL changes to ZYX

Expected: 

URL to be CBA

<hr>

Also, there is a difference between `/root(dialog:other)` and `/root/(dialog:other)`. In the first example, they are siblings. In the second example (with the `/` in front of the `(`), the dialog is a child.

**Comment by Epenance** [comment]
@jasonaden  Sorry for the confussion, it is indeed a bit complex.

To narrow it down as clear as possible:

1. Go to: https://angular-3rhhpw.stackblitz.io
2. Click on "Overlay 1"

Expected:
The text "view1" should stay within the primary outlet, but it disappears.

If you go to: https://angular-3rhhpw.stackblitz.io/list2 and click "Overlay 1" you will notice that it stays which is the expected behaviour.

**Comment by jasonaden** [comment]
Okay, got it now. Thanks for the clarification!

**Comment by atscott** [comment]
confirmed in v9: https://stackblitz.com/edit/angular-3rhhpw?file=app%2Fapp.module.ts
Had to remove overlay in the demo to get it to work in v9, but the bug still exists using the same steps to reproduce:

From root route,
> Click on "Overlay 1"
Expected:
The text "view1" should stay within the primary outlet, but it disappears.

> If you go to: https://angular-3rhhpw.stackblitz.io/list2 and click "Overlay 1" you will notice that it stays which is the expected behaviour.

**Comment by atscott** [comment]
Closing as duplicate of #16051 since the only solution would be to preserve the trailing slash since there needs to be a place to put the auxiliary route, and the URL cannot be `angular-3rhhpw.stackblitz.io(dialog:view1)` so it must by necessity be `angular-3rhhpw.stackblitz.io/(dialog:view1)`
